crates/floresta-chain/src/pruned_utreexo/chainparams.rs

```diff
@@ -52,26 +45,27 @@ pub struct ChainParams {
     pub exceptions: HashMap<BlockHash, c_uint>,
 }
 impl ChainParams {
-    fn max_target(net: Network) -> Uint256 {
+    fn max_target(net: Network) -> Target {
         match net {
-            Network::Bitcoin => max_target(net.into()),
-            Network::Testnet => Uint256([
-                0x0000000000000000,
-                0x0000000000000000,
-                0x0000000000000000,
-                0x00000000ffff0000,
+            Network::Bitcoin => Target::from_be_bytes([
+                0x00, 0x00, 0x00, 0x00, 0xff, 0xff, 0x00, 0x00, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+                0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+                0xff, 0xff, 0xff, 0xff,
             ]),
-            Network::Signet => Uint256([
-                0x0000000000000000,
-                0x0000000000000000,
-                0x0000000000000000,
-                0x00000377ae000000,
+            Network::Testnet => Target::from_be_bytes([
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0xff, 0xff, 0x00, 0x00,
             ]),
-            Network::Regtest => Uint256([
-                0xffffffffffffffff,
-                0xffffffffffffffff,
-                0xffffffffffffffff,
-                0x7fffffffffffffff,
+            Network::Signet => Target::from_be_bytes([
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x03, 0x77,
+                0xae, 0x00, 0x00, 0x00,
+            ]),
+            Network::Regtest => Target::from_be_bytes([
+                0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+                0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x7f, 0xff, 0xff, 0xff,
+                0xff, 0xff, 0xff, 0xff,
```

Comment:
Why do we need to change the type here?

Reply:
`rust-bitcoin` stopped exposing Uint256, and now they strongly type `Work` and `Target`